corporate/tests/test_stripe.py

```diff
@@ -5775,6 +5775,42 @@ def test_invoice_plan_without_stripe_customer(self) -> None:
             "Customer has a paid plan without a Stripe customer ID:",
         )
 
+    @mock_stripe()
+    def test_validate_licenses_for_manual_plan_management(self, *mocks: Mock) -> None:
+        user = self.example_user("hamlet")
+        self.login_user(user)
+
+        with time_machine.travel(self.now, tick=False):
+            self.upgrade(invoice=True, licenses=100)
+
+        with time_machine.travel(self.now, tick=False):
+            result = self.client_billing_patch(
+                "/billing/plan",
+                {"licenses_at_next_renewal": 30},
+            )
+            self.assert_json_success(result)
+
+        # Check number of users in the realm.
+        self.assertEqual(get_realm("zulip").get_active_users().count(), 8)
+
+        # After adding 30 more users, the realm will have 38 users.
+        for count in range(30):
+            do_create_user(
+                f"email {count}",
+                f"password {count}",
+                get_realm("zulip"),
+                "name",
+                acting_user=None,
+            )
```

Comment:
Creating 30 users is pretty expensive; this test will be fairly slow. We should probably change the numbers to create like 2.

Reply:
Or maybe better, convert a few users to guests? That's very cheap to do, comparatively.

Reply:
I simplified this test so that we only add one extra user. I did have to set the minimum licenses to be lower than the constant that's set in the billing code (which isn't in use in production).